Extension/src/LanguageServer/copilotCompletionContextProvider.ts

```diff
@@ -267,21 +306,36 @@ export class CopilotCompletionContextProvider implements ContextResolver<CodeSni
                     copilotCompletionContext.snippets = [];
                 }
             }
+            telemetry.addSnippetCollectionStatistics(copilotCompletionContext?.statistics);
             telemetry.addCompletionContextKind(copilotCompletionContextKind);
             // Handle cancellation.
             if (copilotCompletionContextKind === CopilotCompletionKind.Canceled) {
                 const duration: number = CopilotCompletionContextProvider.getRoundedDuration(resolveStartTime);
-                telemetry.addInternalCanceled(duration);
-                throw new CancellationError();
+                telemetry.addCopilotCanceled(duration);
+                throw new CopilotCancellationError();
             }
             telemetry.addCompletionContextKind(copilotCompletionContextKind);
             telemetry.addResponseMetadata(copilotCompletionContext?.isResultMissing ?? true,
                 copilotCompletionContext?.snippets?.length, copilotCompletionContext?.translationUnitUri,
                 copilotCompletionContext?.caretOffset, copilotCompletionContext?.featureFlag);
+            logMessage += ` (id:${copilotCompletionContext?.requestId}) `;
+            if (!copilotCompletionContext?.snippets?.length) {
+                telemetry.addCompletionContextKind(CopilotCompletionKind.FallbackToSimilarFiles);

```

Comment:
I'm seeing it FallbackToSimilarFiles even when the feature flag is not set?

Reply:
Is there anything I should fix? Let me know :)